File: dev-docs/snippets/rule-sets/update.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Update a rule set's name, description, or rules via typed `UpdateRuleSetParams`.
export async function update(client: JsBaoClient, ruleSetId: string) {
  // #region example
  const ruleSet = await client.ruleSets.update(ruleSetId, {
    description: "Updated policy",
  });
  // #endregion example
  return ruleSet;
}

File: dev-docs/snippets/rule-sets/update.swift
import JsBaoClient

// Update a rule set's name, description, or rules via typed `UpdateRuleSetParams`.
func update(client: JsBaoClient, ruleSetId: String) async throws {
  // #region example
  let ruleSet = try await client.ruleSets.update(
    ruleSetId: ruleSetId,
    params: UpdateRuleSetParams(description: "Updated policy")
  )
  // #endregion example
  _ = ruleSet
}
